### ISSUE TITLE
Crash: MTLTextureDescriptor has invalid pixelFormat

### ISSUE THREAD

**Issue by bytepoets-jhe** [flutter/flutter#145147]
### Steps to reproduce

Device: iPadAir with iOs 12.5.7 (Same Error reported for iPhone 5s on sentry)
Flutter Version: 3.13.8

Open new page with Widgets containing an AssetImage
App crashes

### Expected results

Page should open

### Actual results

App crashes

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
Exception Type: EXC_CRASH (SIGABRT)
Crashed Thread: 4

Application Specific Information:
-[MTLTextureDescriptorInternal validateWithDevice:] > -[MTLTextureDescriptorInternal validateWithDevice:]:1094: failed assertion `MTLTextureDescriptor has invalid pixelFormat (554).'
 > : failed assertion `%s'
 > DRHT > error > MTLTextureDescriptor has invalid pixelFormat (554).

Thread 4 Crashed:
0   libsystem_kernel.dylib          0x3ac39d0dc         __pthread_kill
1   libsystem_pthread.dylib         0x3ac4b0094         <redacted>
2   libsystem_c.dylib               0x3ac216ea8         abort
3   libsystem_c.dylib               0x3ac1e4afc         __assert_rtn
4   Metal                           0x3b0ea0134         MTLReportFailure
5   Metal                           0x3b0e6f1e0         <redacted>
6   AGXMetalA7                      0x3ed9a71e4         <redacted>
7   Flutter                         0x1050a31c8         <redacted>
8   Flutter                         0x1050ee738         <redacted>
9   Flutter                         0x104e3bf1c         <redacted>
10  Flutter                         0x1050eccd8         <redacted>
11  Flutter                         0x104e3a6fc         <redacted>
12  Flutter                         0x104e3dcec         <redacted>
13  CoreFoundation                  0x3acb01554         <redacted>
14  CoreFoundation                  0x3acb01284         <redacted>
15  CoreFoundation                  0x3acb00ab8         <redacted>
16  CoreFoundation                  0x3acafba08         <redacted>
17  CoreFoundation                  0x3acafafb4         CFRunLoopRunSpecific
18  Flutter                         0x104e3ddd8         <redacted>
19  Flutter                         0x104e3d44c         <redacted>
20  libsystem_pthread.dylib         0x3ac4b92c0         <redacted>
21  libsystem_pthread.dylib         0x3ac4b9220         _pthread_start

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel stable, 3.13.8, on macOS 14.0 23A344 darwin-x64, locale en-AT)
    ! Warning: `dart` on your path resolves to /usr/local/Cellar/dart/2.17.0/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/jhe/fvm/versions/3.13.8. Consider adding /Users/jhe/fvm/versions/3.13.8/bin to the front of your path.
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] Android Studio (version 2023.1)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.3.4)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.3.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.3.4)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.5)
[✓] Connected device (2 available)
[✓] Network resources

! Doctor found issues in 2 categories.

```

</details>

**Comment by bytepoets-jhe** [comment]
This looks similar to https://github.com/flutter/flutter/issues/133015

**Comment by darshankawar** [comment]
@bytepoets-jhe 
Since you seem to be on 3.13.8 which doesn't seem to have the fix made for https://github.com/flutter/flutter/issues/133015, which is actually available in 3.16 version onwards. So you will need to upgrade your stable version and run your scenario again to see the fix working.
